services/google_scraping_service/db_updater.go

```diff
@@ -0,0 +1,53 @@
+package google_scraping_service
+
+import (
+	"encoding/json"
+
+	"github.com/gutakk/go-google-scraper/models"
+)
+
+func UpdateKeywordStatus(keywordID uint, status models.KeywordStatus, failedReason error) error {
+	var keywordModel models.Keyword
+	keywordModel.Status = status
+
+	if failedReason != nil {
+		keywordModel.FailedReason = failedReason.Error()
+	}
+
+	err := models.UpdateKeywordByID(keywordID, keywordModel)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func UpdateKeywordWithParsingResult(keywordID uint, parsingResult ParsingResult) error {
+	nonAdwordLinks, nonAdwordLinksParsingErr := json.Marshal(parsingResult.NonAdwordLinks)
+	if nonAdwordLinksParsingErr != nil {
```

Comment:
Because `NonAdwordLinks` and `TopPositionAdwordLinks` type are `datatypes.JSON` from `gorm.io/datatypes` and the datatypes.JSON actual type is `[]byte`

Reply:
Alright. Thanks for the explanation. I did miss the JSON data type for these columns 🙈